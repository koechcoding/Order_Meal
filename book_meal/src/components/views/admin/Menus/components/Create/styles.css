.Select-control:active,
.Select-control:focus,
.Select-control {
  border-radius: 0;
  background-color: #f4f1d6;
  outline: none; 
}

.Select-placeholder {
  color: #666; 
}