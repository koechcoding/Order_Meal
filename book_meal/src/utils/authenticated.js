export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}