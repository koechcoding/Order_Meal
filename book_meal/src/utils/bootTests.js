import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
