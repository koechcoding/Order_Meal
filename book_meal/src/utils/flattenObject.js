export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}