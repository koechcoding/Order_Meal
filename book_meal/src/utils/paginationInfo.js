export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}