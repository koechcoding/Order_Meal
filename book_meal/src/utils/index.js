import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}