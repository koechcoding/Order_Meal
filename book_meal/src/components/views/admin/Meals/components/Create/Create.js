import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import { singleError } from 'src/utils';
import Modal from 'src/components/common/Modal';
import { Alert, Input, Button } from 'reactstrap';
import ImageInput from 'src/components/common/ImageInput';
import { IMAGES_UPLOAD_URL, IMAGE_UPLOAD_PRESET } from 'src/constants';

class CreateModal extends React.Component {

    state = {
        name: '',
        cost: '',
    }

    reset = () => {
        setTimeout(() => {
            this.setState({
                ...this.state,
                success: false,
                error: null
            });
        }, 2000);
    }

    onImageAdded = (data) => {
        this.setState({
            ...this.state,
            image: data,
        });
    }

    onOpened = () => {
        this.setState({
            ...this.state,
            error: null,
            success: false,
            image: null,
        });
    }

    onImageRemoved = () => {
        this.setState({
            ...this.state,
            image: null,
        });
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        })
    }

    onCreate = () => {
        // on success...
        const resolve = () => {
            // reset headers
            axios.auth();
            this.setState({
                ...this.state,
                success: true,
                error: null,
                name: '',
                cost: '',
                image: null,
            });
            this.props.onChange();
            this.reset();
        }

        // on failure...
        const reject = (response) => {
            this.setState({
                ...this.state,
                error: response,
                success: false,
            });
            this.reset();
            this.props.setLoading(false);
        }
        this.props.setLoading(true);

        const { image, name, cost } = this.state;
        if (image) {
            const imageUpload = {
                file: image,
                upload_preset: IMAGE_UPLOAD_PRESET
            };

            // first upload the image
            delete axios.defaults.headers.common.Authorization;
            delete axios.defaults.headers.common['Access-Control-Allow-Origin'];
            axios.post(IMAGES_UPLOAD_URL, imageUpload).then(({ data }) => {
                axios.auth();
                axios.post('/meals', { 
                    name, 
                    cost, 
                    img_url: data.secure_url 
                }).then(() => {
                    resolve();
                }).catch(({ response }) => {
                    reject(response)
                });

            }).catch(({ response }) => {
                reject(response);
            });
        } else {
            axios.post('/meals', { name, cost }).then(() => {
                resolve();
            }).catch(({ response }) => {
                reject(response);
            })
        }
    }

    render() {
        const { error, success } = this.state;
        const body = (
            <div>
                {success &&
                    <Alert className="text-center text-small" color="success">
                        Successfully added
                    </Alert>
                }
                {error &&
                    <Alert className="text-center text-small" color="danger">
                        { singleError(error) }
                    </Alert>
                }

                <ImageInput 
                    onImageAdded={this.onImageAdded} 
                    onImageRemoved={this.onImageRemoved}
                />
                <div className="pl-4 pr-4">
                    <label> Name </label>
                    <Input name="name" value={this.state.name} onChange={this.onChange} type="text" />
                    <label> Cost </label>
                    <Input name="cost" value={this.state.cost} onChange={this.onChange} type="number" />
                </div>
            </div>
        );

        const footer = (
            <Button color="primary" className="m-auto"onClick={this.onCreate}>Save Meal</Button>
        )
        const { isOpen, toggle } = this.props;

        return (
            <Modal 
                title="Add New Meal" 
                body={body} 
                footer={footer} 
                isOpen={isOpen}
                toggle={toggle}
                onOpened={this.onOpened}
            />
        );
    }

}

CreateModal.propTypes = {
    toggle: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    setLoading: PropTypes.func.isRequired,
    isOpen: PropTypes.bool,

}

export default CreateModal;