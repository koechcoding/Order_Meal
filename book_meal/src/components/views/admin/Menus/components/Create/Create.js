import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button, Input } from 'reactstrap';
import { singleError } from 'src/utils';
import { Async } from 'react-select';
import _ from 'lodash';

import 'react-select/dist/react-select.css';
import './styles.css';

class CreateModal extends React.Component {
    state = {
        quantity: 1,
    }

    reset=()=>{
        this.setState({
            ...this.state,
            meal: "",
            menu: "",
        });
        setTimeout(()=>{
            this.setState({
                ...this.state,
                success: false,
                error: null
            });
        }, 2000);
    }

    onOpened=()=>{
        this.setState({
            ...this.state,
            error: null,
            success: false,
            menus: [],
            meals: [],
        })
    }

    onChange=(e)=>{
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        })
    }
}