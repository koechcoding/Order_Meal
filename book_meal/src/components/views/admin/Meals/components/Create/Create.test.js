import React from 'react';
import { shallow, mount }  from 'enzyme';
import 'src/utils/bootTests'; 
import Create from './Create';
import Adapter from 'enzyme-adapter-react-16';

const props = { toggle:() => {}, onChange:() => {}, setLoading:() => {}}
describe('<Create />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Create {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('adds image', () => {
        const wrapper = shallow(<Create {...props} />);
        wrapper.instance().onImageAdded('test');
        expect(wrapper.state('image')).toBe('test');
    });
    it('resets on opened', () => {
        const wrapper = shallow(<Create {...props} />);
        wrapper.setState({...wrapper.state(), error: true, success: true});
        wrapper.instance().onOpened();
        expect(wrapper.state('error')).toBe(null);
        expect(wrapper.state('success')).toBe(false);
    });
    it('removes image', () => {
        const wrapper = shallow(<Create {...props} />);
        wrapper.instance().onImageRemoved();
        expect(wrapper.state('image')).toBe(null);
    });
    it('records input change', () => {
        const wrapper = mount(<Create {...props} />);
        wrapper.instance().onChange({target: {name: 'name', value: 'value'}});
        expect(wrapper.state('name')).toBe('value');
    });
});
