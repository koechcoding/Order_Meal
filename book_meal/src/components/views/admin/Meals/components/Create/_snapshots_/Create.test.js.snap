/ Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateModal
    onChange={[Function]}
    setLoading={[Function]}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "body": <div>
        <ImageInput
                onImageAdded={[Function]}
                onImageRemoved={[Function]}
        />
        <div
                className="pl-4 pr-4"
        >
                <label>
                         Name 
                </label>
                <Input
                        name="name"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
                <label>
                         Cost 
                </label>
                <Input
                        name="cost"
                        onChange={[Function]}
                        type="number"
                        value=""
                />
        </div>
</div>,
      "footer": <Button
        className="m-auto"
        color="primary"
        onClick={[Function]}
        tag="button"
>
        Save Meal
</Button>,
      "isOpen": undefined,
      "onOpened": [Function],
      "title": "Add New Meal",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": <div>
          <ImageInput
                    onImageAdded={[Function]}
                    onImageRemoved={[Function]}
          />
          <div
                    className="pl-4 pr-4"
          >
                    <label>
                               Name 
                    </label>
                    <Input
                              name="name"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
                    <label>
                               Cost 
                    </label>
                    <Input
                              name="cost"
                              onChange={[Function]}
                              type="number"
                              value=""
                    />
          </div>
</div>,
        "footer": <Button
          className="m-auto"
          color="primary"
          onClick={[Function]}
          tag="button"
>
          Save Meal
</Button>,
        "isOpen": undefined,
        "onOpened": [Function],
        "title": "Add New Meal",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;